### PR TITLE
Multishop - BO : Fix orders not appearing in order list for shared space shops

### LINKED ISSUE #17727: I should be able to see orders placed in other shops of the same group, if the group shares orders
**Describe the bug**
I should be able to see orders placed in other group stores, if the group shares orders

**To Reproduce**
1. Go to Shop parameters > General and enable multistore
2. Go to Advanced parameters > Multistore
3. Click on Add new shop group "all shared group"
4. Share customers, share available quantities to sell, share orders
5. Create a new store "shop3" in "all shared group"
6. Create a new store "shop4" in "all shared group"
7. Make an order in FO in shop3
8. Go to orders > orders migrated listing
9. Select shop4
10. You should see the order make in shop3 because orders are shared in this group

**Screenshots**
capture d'écran_2364

capture d'écran_2365

**Additional information**
PrestaShop version: 1.7.7.x
PHP version: N/A

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | In multishop, when two shops share the same space and the space is configured to share orders: an order made on one shop should appear on the other in the order list in the BO. See issue for more detailed explanations.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #17727
| How to test?  | **Copy pasted from the issue:**<br>- Go to Shop parameters > General and enable multistore<br> - Go to Advanced parameters > Multistore<br> - Click on Add new shop group "all shared group"<br> - Share customers, share available quantities to sell, share orders<br> - Create a new store "shop3" in "all shared group"<br> - Create a new store "shop4" in "all shared group"<br>- Make an order in FO in shop3<br> - Go to orders > orders **migrated** listing<br> - Select shop4<br> - You should see the order make in shop3 because orders are shared in this group

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17767)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml [hunk 1]
```diff
@@ -206,7 +206,7 @@ services:
             - '%database_prefix%'
             - '@prestashop.core.query.doctrine_search_criteria_applicator'
             - '@=service("prestashop.adapter.legacy.context").getContext().language.id'
-            - '@=service("prestashop.adapter.shop.context").getContextListShopID()'
+            - '@=service("prestashop.adapter.shop.context").getContextListShopID("share_order")'
```
Comment (matthieu-rolland) [inline src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml:210]: Here "share_order" is the value of a constant in the Shop ObjectModel: `const SHARE_ORDER = 'share_order';`

I'm sure there must be a way not to hardcode it in the yml but I don't know how 😅  Any idea ?

Comment (PierreRambaud) [reply src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml:210]: You can have a look at https://symfony.com/blog/new-in-symfony-3-2-php-constants-in-yaml-files :)

Comment (matthieu-rolland) [reply src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml:210]: thanx !

**Review by matks — COMMENTED** [review]
Same feedback than Pierre; else code is 👍

File: src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml (hunk 1 above)
Comment (matthieu-rolland) [reply src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml:210]: @PierreRambaud 

Unfortunately I cannot use this directly in a method parameter.

I also tried putting the class constant in a yaml parameter, and using it as a parameter for the getContextListShopID method, it doesn't work either...

I see no other way than to hardcode it for now...

Comment (matthieu-rolland) [reply src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml:210]: @PierreRambaud ok with @matks we found that we have to use the "parameter()" method 👍

**Review by atomiix — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @matthieu-rolland

**Pull request merged**